sycl/plugins/cuda/pi_cuda.cpp

```diff
@@ -163,6 +163,8 @@ pi_result check_error(CUresult result, const char *function, int line,
 
 /// \cond NODOXY
 #define PI_CHECK_ERROR(result) check_error(result, __func__, __LINE__, __FILE__)
+// Use (void) to silence unused warnings.
+#define assertm(exp, msg) assert(((void)msg, exp))
```

Comment:
I am not sure I understand the need for this. The case where it is used do not seem have any potentially unused variables, and if the string is a problem, we normally just use something like `assert(predicate && "YOUR STRING HERE");`.

Reply:
Thanks, I was exactly trying for predicate + a string . Will fix that.